tests/integration/bond_test.py

```diff
@@ -321,23 +324,6 @@ def test_set_bond_mac_address(eth1_up):
         assert_mac_address(current_state, MAC1)
 
 
-def test_reordering_the_slaves_does_not_change_the_mac(bond99_with_2_slaves):
```

Comment:
Please keep this test and also cover other scenarios from:

Ok, so the expectation is that the MAC address will not change for these these changes of the order of the slave NICs in the desired state:

Adding/removing:

[eth2] -> [eth1, eth2]
[eth1, eth2, eth3] -> [eth2, eth3]
Replacing:

[eth1, eth2] -> [eth3]
Reordering:

[eth1, eth2] -> [eth2, eth1]

Reply:
Yes, I was wrong to mention this scenario should be removed from coverage. My fault.
This test is checking the same but with the profiles on the system... this is a different scenario.

Regarding other tests, I think they are needed but do not belong to this change.

Reply:
The test is back. I am going to work on the other test cases.

Reply:
Forgot two scenarios:

[eth1] -> []
[] -> [eth1]